pkg/plugins/manager/loader/loader.go

```diff
@@ -81,6 +84,39 @@ func (l *Loader) Load(ctx context.Context, class plugins.Class, paths []string)
 	return l.loadPlugins(ctx, class, pluginJSONPaths)
 }
 
+func (l *Loader) createPluginsForLoading(class plugins.Class, foundPlugins foundPlugins) map[string]*plugins.Plugin {
```

Comment:
This was extracted from `l.loadPlugins` because golangci-lint was complaining:

```
pkg/plugins/manager/loader/loader.go:87:1: cyclomatic complexity 31 of func `(*Loader).loadPlugins` is high (> 30) (gocyclo)
func (l *Loader) loadPlugins(ctx context.Context, class plugins.Class, pluginJSONPaths []string) ([]*plugins.Plugin, error) {
^
```

A similar refactoring should probably be done for the other parts of `loadPlugins` as well (in another PR)